compiler/ballerina-lang/src/main/java/io/ballerina/projects/directory/BuildProject.java

```diff
@@ -155,6 +155,13 @@ private BuildProject(ProjectEnvironmentBuilder environmentBuilder, Path projectP
         return Optional.empty();
     }
 
+    @Override
+    public Project duplicate() {
+        BuildProject buildProject = new BuildProject(
```

Comment:
Does this work for situations where I have created the environment with a non-default builder?

Reply:
No, for the moment we support only with the DefaultEnvironmentBuilder as we don't have use cases for other environments.

Reply:
Can we create an issue to capture this limitation? I am pretty sure we are going to encounter this sooner or later.